libs/sdk-core/src/models.rs

```diff
@@ -982,6 +982,37 @@ pub struct SendOnchainResponse {
     pub reverse_swap_info: ReverseSwapInfo,
 }
 
+/// See [ReverseSwapFeesRequest]
+pub struct PrepareOnchainPaymentRequest {
+    pub amount_sat: u64,
+
+    /// If true, amount is send_amount. If false, amount is receive_amount.
+    pub is_send_amount: bool,
```

Comment:
Did you try different ways to express that? Perhaps enum? Or two optional fields (send_amount, received_amount)?
If enum with two variants translates simply to foreign languages then it could be a better option.

Reply:
Something like this? 0f926038f03839474c3f95e3006f31fab8b3b545

I initially avoided enums, because AFAIK foreign languages need named inner fields, and that would become verbose. But using a plain enum instead of the boolean is a good solution IMO.